Library/Homebrew/extend/os/mac/requirements/swift_requirement.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require "requirements/xcode_requirement"
+
+class SwiftRequirement < Requirement
+  satisfy build_env: false do
+    next unless @version
+
+    @xcode_requirement = XcodeRequirement.new([xcode_required_version, *tags])
+    @xcode_requirement.satisfied?
+  end
+
+  def xcode_required_version
```

Comment:
Might be nice to keep this in `Library/Homebrew/os/mac/xcode.rb` so it's updated when that is.

Reply:
Actually I put it there initially. I moved it out because this goes opposite direction which would make it only reverse lookup there but if you are ok with that I'll put it back 🤷

Reply:
I think it's fine for the lookup to be the same way around as here but it's more likely to get bumped there, I think.